### PR TITLE
COTECH-765 | Anyclip updates for gamefaqs

### PR DESCRIPTION
**Pull request by latata**
The idea is to use the same approach for anyclip as we use in FCP. So the player container is added by adengine using `slotCreatorConfig`. This way we will be less dependant on gamefaqs repo and we will be able to make changes in anyclip for gamefaqs just in adegine repo

### CODE REVIEW

File: src/platforms/news-and-ratings/shared/setup/dynamic-slots/news-and-ratings-anyclip.setup.ts [hunk 1]
```diff
@@ -29,11 +29,8 @@ export class NewsAndRatingsAnyclipSetup implements DiProcess {
 			const doesAnyclipTagExist = context.get('services.anyclip.anyclipTagExists');
 			const pathname = window.location.pathname.toLowerCase();
 
-			const isApplicable = doesAnyclipTagExist
-				? doesAnyclipTagExist
-				: this.isApplicable(pname)
-				? this.isApplicable(pname)
-				: this.isApplicable(pathname);
+			const isApplicable =
```
Comment (latata) [inline src/platforms/news-and-ratings/shared/setup/dynamic-slots/news-and-ratings-anyclip.setup.ts:32]: the logic is the same - just cleaner

File: src/platforms/news-and-ratings/shared/setup/dynamic-slots/news-and-ratings-dynamic-slots.setup.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ export class NewsAndRatingsDynamicSlotsSetup implements DiProcess {
 		context.set('state.topLeaderboardExists', tlbExists);
 
 		if (!tlbExists) {
-			communicationService.on(eventsRepository.AD_ENGINE_STACK_START, () => {
+			communicationService.on(eventsRepository.AD_ENGINE_STACK_COMPLETED, () => {
```
Comment (latata) [inline src/platforms/news-and-ratings/shared/setup/dynamic-slots/news-and-ratings-dynamic-slots.setup.ts:65]: new event is used to make sure the ad slot is already created.

**Review by dzawada — DISMISSED** [review]

File: src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { GameFAQsSlotsDefinitionRepository } from './gamefaqs-slots-definition-repository';
+
+@Injectable()
+export class GameFAQsAnyclipSetup implements DiProcess {
+	constructor(private slotsDefinitionRepository: GameFAQsSlotsDefinitionRepository) {}
+
+	execute(): void {
+		context.set('custom.hasIncontentPlayer', !!document.querySelector('.msg_body'));
+
+		// TODO: this line can be removed after releasing gamefaqs change that removes the element from the page
```
Comment (nandy-andy) [inline src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts:13]: Shall we fill a follow-up ticket for that or you plan not to close 765 until it's removed?

**Review by nandy-andy — DISMISSED** [review]

File: src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { context, DiProcess } from '@wikia/ad-engine';
+import { Injectable } from '@wikia/dependency-injection';
+import { GameFAQsSlotsDefinitionRepository } from './gamefaqs-slots-definition-repository';
+
+@Injectable()
+export class GameFAQsAnyclipSetup implements DiProcess {
+	constructor(private slotsDefinitionRepository: GameFAQsSlotsDefinitionRepository) {}
+
+	execute(): void {
+		context.set('custom.hasIncontentPlayer', !!document.querySelector('.msg_body'));
```
[the remark this answers is not in the document]
Comment (latata) [reply src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts:11]: I've checked the usage of this class in gamefaqs and it seems to be used only there

File: src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts (hunk 3 above)
Comment (latata) [reply src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts:13]: I've created a PR so we can handle it within the scope of the current ticket.

Comment (latata) [reply src/platforms/news-and-ratings/gamefaqs/setup/context/slots/gamefaqs-anyclip.setup.ts:13]: https://github.com/Wikia/ad-engine/pull/2226/files

**Review by nandy-andy — APPROVED** [review]

**Review by dzawada — APPROVED** [review]

**Pull request merged**